shell/server/proxy.js

```diff
@@ -1549,6 +1597,10 @@ class Proxy {
         response.setHeader("Set-Cookie", rpcResponse.setCookies.map(makeSetCookieHeader));
       }
 
+      const mainUrl = process.env.ROOT_URL;
+      response.setHeader("Content-Security-Policy", "frame-ancestors " + mainUrl);
```

Comment:
We use Meteor's BrowserPolicy API to set it. The difficulty here is that `child-src` is unsupported by many browsers, though maybe the situation is better now than when this was written...